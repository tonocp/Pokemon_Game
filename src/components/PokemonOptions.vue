<template>
  <div class="options-container">
    <ul>
      <li v-for="pokemon in pokemons" :key="pokemon.id">{{ pokemon.name.toUpperCase() }}</li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    pokemons: {
      type: Array,
      required: true,
    },
  },
};
</script>

<style scoped>
ul {
  list-style-type: none;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  padding: 15px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
