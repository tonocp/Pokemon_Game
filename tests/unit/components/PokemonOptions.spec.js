import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";
import { pokemons } from "../mocks/pokemons.mock";

describe("PokemonOptions component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons,
      },
    });
  });

  test("Debe hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Debe mostrar las 4 opciones correctamente", () => {
    const liTags = wrapper.findAll("li");

    expect(liTags.length).toBe(4);
    expect(liTags[0].text()).toBe("BULBASAUR");
    expect(liTags[1].text()).toBe("IVYSAUR");
    expect(liTags[2].text()).toBe("VENUSAUR");
    expect(liTags[3].text()).toBe("CHARMANDER");
  });

  test("Debe emitir 'selection' con sus respectivos parÃ¡metros al hacer click", () => {
    const [li1, li2, li3, li4] = wrapper.findAll("li");

    li1.trigger("click");
    li2.trigger("click");
    li3.trigger("click");
    li4.trigger("click");

    expect(wrapper.emitted("selection").length).toBe(4);
    expect(wrapper.emitted("selection")[0]).toEqual([1]);
    expect(wrapper.emitted("selection")[1]).toEqual([2]);
    expect(wrapper.emitted("selection")[2]).toEqual([3]);
    expect(wrapper.emitted("selection")[3]).toEqual([4]);
  });
});
