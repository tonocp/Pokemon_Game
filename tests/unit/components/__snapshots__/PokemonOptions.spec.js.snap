// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component Debe hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>BULBASAUR</li>
    <li>IVYSAUR</li>
    <li>VENUSAUR</li>
    <li>CHARMANDER</li>
  </ul>
</div>
`;
